common/config/clickhouse.go

```diff
@@ -24,20 +30,57 @@ func (c *Clickhouse) ModifyConfig(dest ExporterConfigurer, currentConfig *Config
 		return errors.New("clickhouse endpoint not specified, gateway will not be configured for Clickhouse")
 	}
 
-	username, userExists := dest.GetConfig()[clickhouseUsername]
-	password, passExists := dest.GetConfig()[clickhousePassword]
-	if userExists != passExists {
-		return errors.New("clickhouse username and password must be both specified, or neither")
+	if !strings.Contains(endpoint, "://") {
+		endpoint = "tcp://" + endpoint
+	}
+
+	parsedUrl, err := url.Parse(endpoint)
+	if err != nil {
+		return errors.New("clickhouse endpoint is not a valid URL")
+	}
+
+	if parsedUrl.Port() == "" {
+		endpoint = strings.Replace(endpoint, parsedUrl.Host, parsedUrl.Host+":9000", 1)
 	}
 
+	username, userExists := dest.GetConfig()[clickhouseUsername]
+
 	exporterName := "clickhouse/clickhouse-" + dest.GetID()
 	exporterConfig := GenericMap{
 		"endpoint": endpoint,
 	}
 	if userExists {
 		exporterConfig["username"] = username
-		exporterConfig["password"] = password
+		exporterConfig["password"] = "${CLICKHOUSE_PASSWORD}"
+	}
+
+	createSchema, exists := dest.GetConfig()[clickhouseCreateSchema]
+	createSchemaBoolValue := exists && strings.ToLower(createSchema) == "create"
+	exporterConfig["create_schema"] = createSchemaBoolValue
+
+	dbName, exists := dest.GetConfig()[clickhouseDatabaseName]
+	if !exists {
+		return errors.New("clickhouse database name not specified, gateway will not be configured for Clickhouse")
+	}
+	exporterConfig["database"] = dbName
+
+	tracesTable, exists := dest.GetConfig()[clickhouseTracesTable]
+	if !exists {
+		return errors.New("clickhouse traces table not specified, gateway will not be configured for Clickhouse")
```

Comment:
I don't think we should return an error here. It is legitimate to have a table only for metrics or logs. right?